google/cloud/aiplatform/models.py

```diff
@@ -345,589 +468,768 @@ def deploy(
+        """Helper method to instantiates prediction client for this endpoint.
 
-        self._job_client = initializer.global_config.create_client(
-            client_class=job_service.JobServiceClient,
+        Args:
+            location (str): The location of this endpoint.
+            credentials (google.auth.credentials.Credentials):
+                Optional custom credentials to use when accessing interacting with
+                the prediction client.
+        Returns:
+            prediction_client (prediction_service_client.PredictionServiceClient):
+                Initalized prediction client.
+        """
+        return initializer.global_config.create_client(
+            client_class=prediction_service_client.PredictionServiceClient,
             credentials=credentials,
-            location_override=self.location,
+            location_override=location,
+            prediction_client=True,
         )
 
-        # Make blocking call to service
-        gapic_batch_prediction_job = self._job_client.create_batch_prediction_job(
-            request=create_batch_prediction_job_request
-        )
+    def predict(self, instances: List, parameters: Optional[Dict] = None) -> Prediction:
+        """Make a prediction against this Endpoint.
 
-        # Get name of new BatchPredictionJob and return SDK representation
-        new_batch_prediction_job_name = gapic_batch_prediction_job.name
+        Args:
+            instances (List):
+                Required. Required. The instances that are the input to the
+                prediction call. A DeployedModel may have an upper limit
+                on the number of instances it supports per request, and
+                when it is exceeded the prediction call errors in case
+                of AutoML Models, or, in case of customer created
+                Models, the behaviour is as documented by that Model.
+                The schema of any single instance may be specified via
+                Endpoint's DeployedModels'
+                [Model's][google.cloud.aiplatform.v1beta1.DeployedModel.model]
+                [PredictSchemata's][google.cloud.aiplatform.v1beta1.Model.predict_schemata]
+                ``instance_schema_uri``.
+            parameters (Dict):
+                The parameters that govern the prediction. The schema of
+                the parameters may be specified via Endpoint's
+                DeployedModels' [Model's
+                ][google.cloud.aiplatform.v1beta1.DeployedModel.model]
+                [PredictSchemata's][google.cloud.aiplatform.v1beta1.Model.predict_schemata]
+                ``parameters_schema_uri``.
+        Returns:
+            prediction: Prediction with returned predictions and Model Id.
 
-        return jobs.BatchPredictionJob(
-            batch_prediction_job_name=new_batch_prediction_job_name
+        """
+        prediction_response = self._prediction_client.predict(
```

Comment:
I noticed that this raises the following when the training job is not finished when this is called:

AttributeError: 'Endpoint' object has no attribute '_prediction_client'

A check for pending futures would be nice.

Reply:
I'll add that. I considered this earlier but there's a tradeoff because the predict API can be latency sensitive for some use cases.